### PR TITLE
Questionnaire bug fixes | extraction only on completed

### LINKED ISSUE #1700: [Quest] Update Extraction triggering to only happen for QuestionnaireResponses with the `status` == `Completed`
**Enhancement context**
- We currently, don't check the status of a QuestionnaireResponse before we trigger extraction of any kind. 
- We should only trigger extraction of schedules or to different resources when the Questionnaire is completely filled and the QuestionnaireResponse status is marked as completed. 

**Implementation**
- [x] Add the complete status on the QuestionnaireResponse when the `SAVE` is clicked on the questionnaire. This status should only be added when the QR is not marked as draft. 
- [x] Attach the QR subject before the extraction bit. This should be added immediately after the QR is created and not during the extraction period.  
- [x] Check the QuestionnaireResponse status before passing it through the `extraction` method. 
    - [x] The QuestionnaireResponse status should be set to `Completed` for the extraction to be triggered.  
- [x] If the QuestionnaireResponse status is not `completed` we save the resource as is to the resources table on SQLite. 

**Acceptance Criteria**
- [ ] Extraction of any kind should only happen when the QuestionnaireResponse status is set to `Completed`

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes #1700 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by ellykits — COMMENTED** [review]
Looks good. Can you verify all the checklisted acceptance criteria on the issue are all met.

**Comment by maimoonak** [comment]
done!

**Review by ellykits — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 1]
```diff
@@ -247,7 +250,28 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
   fun getQuestionnaireResponse(): QuestionnaireResponse {
     val questionnaireFragment =
       supportFragmentManager.findFragmentByTag(QUESTIONNAIRE_FRAGMENT_TAG) as QuestionnaireFragment
-    return questionnaireFragment.getQuestionnaireResponse()
+    return questionnaireFragment.getQuestionnaireResponse().apply {
+      // TODO this is required until require condition in [QuestionnaireResponseValidator] is fixed
```
Comment (dubdabasoduba) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:254]: @maimoonak for the todo here. Do you have an idea of what needs to be done?  If do we have an issue for it? 

@f-odhiambo Can we add this to the SDK sprint?

**Review by dubdabasoduba — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt (hunk 1 above)
Comment (maimoonak) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:254]: @dubdabasoduba @f-odhiambo https://github.com/google/android-fhir/issues/1713

**Pull request merged**